sdk/core/Azure.Core/src/DynamicData/MutableJsonElement.cs

```diff
@@ -26,10 +27,11 @@ namespace Azure.Core.Json
         private readonly string _path;
         private readonly int _highWaterMark;
 
-        internal const string SerializationRequiresUnreferencedCode = "This method utilizes reflection-based JSON serialization which is not compatible with trimming.";
+        internal const string SerializationRequiresUnreferencedCodeConstructor = "This struct utilizes reflection-based JSON serialization which is not compatible with trimming.";
 
         private readonly MutableJsonDocument.ChangeTracker Changes => _root.Changes;
 
+        [RequiresUnreferencedCode(SerializationRequiresUnreferencedCodeConstructor)]
```

Comment:
I would just add a comment that the default .ctor which is generated by the runtime/compiler should in theory be annotated as well, but the struct is not going to function if it is actually used, so it's OK that it isn't (and really can't be annotated currently).

Reply:
Added a comment